library-model/library-model-generator-integration-test/src/test/java/com/uber/nullaway/libmodel/LibraryModelIntegrationTest.java

```diff
@@ -123,4 +123,32 @@ public void libraryModelInnerClassNullableReturnsTest() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void libraryModelInnerClassNullableUpperBoundsTest() {
+    compilationHelper
+        .setArgs(
+            Arrays.asList(
+                "-d",
+                temporaryFolder.getRoot().getAbsolutePath(),
+                "-XepOpt:NullAway:AnnotatedPackages=com.uber",
+                "-XepOpt:NullAway:JSpecifyMode=true",
+                "-XepOpt:NullAway:JarInferEnabled=true",
+                "-XepOpt:NullAway:JarInferUseReturnAnnotations=true"))
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "import com.uber.nullaway.libmodel.AnnotationExample;",
+            "class Test {",
```

Comment:
I would suggest we also update the `libraryModelWithoutJarInferEnabledTest` test.  But I think we will get errors there even without the modeling due to https://github.com/uber/NullAway/issues/872

Reply:
So write the test libraryModelInnerClassNullableUpperBoundsTest without Jar Infer for a negative case? I tried it and did not see an issue, maybe I didn't understand correctly

Reply:
Without JarInfer enabled, I would expect an error wherever the code has `UpperBoundExample<@Nullable Object>`, since it's only in the model that the type variable for `UpperBoundExample` gets a `@Nullable` upper bound.

Reply:
I understand the issue now, looking into it. Do you think we should fix #872  in a separate PR?

Reply:
Yes I think we can fix #872 separately but we still need to fix the testing here

Reply:
In fact I think #958 will fix #872